imap_processing/tests/hit/test_hit_l1b.py

```diff
@@ -2,51 +2,174 @@
+def hk_dataset(dependencies):
+    """Get the housekeeping dataset"""
+    datasets = hit_l1b.hit_l1b(dependencies, "001")
+    for dataset in datasets:
+        if dataset.attrs["Logical_source"] == "imap_hit_l1b_hk":
+            return dataset
 
 
-def test_create_hk_dataset():
-    """Test creating housekeeping L1B dataset
+def test_hit_l1b_hk_dataset_variables(hk_dataset):
+    """Test the variables in the housekeeping dataset"""
+    # Define the keys that should have dropped from the housekeeping dataset
+    dropped_keys = {
+        "pkt_apid",
+        "sc_tick",
+        "version",
+        "type",
+        "sec_hdr_flg",
+        "seq_flgs",
+        "src_seq_ctr",
+        "pkt_len",
+        "hskp_spare1",
+        "hskp_spare2",
+        "hskp_spare3",
+        "hskp_spare4",
+        "hskp_spare5",
+    }
+    # Define the keys that should be present in the housekeeping dataset
+    valid_keys = {
+        "heater_on",
+        "fsw_version_b",
+        "ebox_m12va",
+        "phasic_stat",
+        "ebox_3d4vd",
+        "ebox_p2d0vd",
+        "temp1",
+        "last_bad_seq_num",
+        "ebox_m5d7va",
+        "ebox_p12va",
+        "table_status",
+        "enable_50khz",
+        "mram_disabled",
+        "temp3",
+        "preamp_l1a",
+        "l2ab_bias",
+        "l34b_bias",
+        "fsw_version_c",
+        "num_evnt_last_hk",
+        "dac1_enable",
+        "preamp_l234b",
+        "analog_temp",
+        "fee_running",
+        "fsw_version_a",
+        "num_errors",
+        "test_pulser_on",
+        "dac0_enable",
+        "preamp_l1b",
+        "l1ab_bias",
+        "l34a_bias",
+        "leak_i",
+        "last_good_cmd",
+        "lvps_temp",
+        "idpu_temp",
+        "temp2",
+        "preamp_l234a",
+        "last_good_seq_num",
+        "num_good_cmds",
+        "heater_control",
+        "hvps_temp",
+        "ebox_p5d7va",
+        "spin_period_long",
+        "enable_hvps",
+        "temp0",
+        "spin_period_short",
+        "dyn_thresh_lvl",
+        "num_bad_cmds",
+        "adc_mode",
+        "ebox_5d1vd",
+        "active_heater",
+        "last_error_num",
+        "last_bad_cmd",
+        "ref_p5v",
+        "code_checksum",
+        "mode",
+    }
+    # Check that the dataset has the correct variables
+    assert valid_keys == set(hk_dataset.data_vars.keys())
+    assert set(dropped_keys).isdisjoint(set(hk_dataset.data_vars.keys()))
 
-    Creates a xarray dataset for housekeeping data
-    """
 
-    # create the attribute manager for this data level
-    attr_mgr = ImapCdfAttributes()
-    attr_mgr.add_instrument_global_attrs(instrument="hit")
-    attr_mgr.add_instrument_variable_attrs(instrument="hit", level="l1b")
-    attr_mgr.add_global_attribute("Data_version", "001")
+def test_hit_l1b_hk_dataset_attributes(hk_dataset):
+    """Test the attributes, dims, and coords in the housekeeping dataset"""
+    # Define the housekeeping dataset attributes
+    dataset_attrs = {
```

Comment:
You can probably just spot-check this since the attribute management should be handled in imap_attribute_manager, you just want to make sure the attributes actually got added properly to your output dataset. IMO it will be annoying to upkeep this test if we ever change, for example, the global attribute variables.

Reply:
Good point. This was mainly to help me with initial validation of attributes for housekeeping, but probably isn't worth keeping in the test file